pkg/api/controller.go

```diff
@@ -5396,6 +5497,96 @@ func (c *Controller) MergeIntoBranch(w http.ResponseWriter, r *http.Request, bod
 	})
 }
 
+func (c *Controller) MergeIntoBranchAsync(w http.ResponseWriter, r *http.Request, body apigen.MergeIntoBranchAsyncJSONRequestBody, repository, sourceRef, destinationBranch string) {
+	if !c.authorize(w, r, permissions.Node{
+		Permission: permissions.Permission{
+			Action:   permissions.CreateCommitAction,
+			Resource: permissions.BranchArn(repository, destinationBranch),
+		},
+	}) {
+		return
+	}
+	ctx := r.Context()
+	c.LogAction(ctx, "merge_branches_async", r, repository, destinationBranch, sourceRef)
+	user, err := auth.GetUser(ctx)
+	if err != nil {
+		writeError(w, r, http.StatusUnauthorized, "user not found")
+		return
+	}
+	metadata := map[string]string{}
+	if body.Metadata != nil {
+		metadata = body.Metadata.AdditionalProperties
+	}
+
+	taskID, err := c.asyncOpsHandler.SubmitMergeIntoBranch(ctx,
+		repository, destinationBranch, sourceRef,
+		user.Committer(),
+		swag.StringValue(body.Message),
+		metadata,
+		swag.StringValue(body.Strategy),
+		graveler.WithForce(swag.BoolValue(body.Force)),
+		graveler.WithAllowEmpty(swag.BoolValue(body.AllowEmpty)),
+		graveler.WithSquashMerge(swag.BoolValue(body.SquashMerge)),
+	)
+
+	if c.handleAPIError(ctx, w, r, err) {
+		return
+	}
+	writeResponse(w, r, http.StatusAccepted, apigen.TaskCreation{
+		Id: taskID,
+	})
+}
+
+func (c *Controller) MergeIntoBranchStatus(w http.ResponseWriter, r *http.Request, repository, sourceRef, destinationBranch string, params apigen.MergeIntoBranchStatusParams) {
```

Comment:
Does this only handle async status response?
If yes, rename API operation to MergeIntoBranchAsyncStatus
If not, e.g it also handles non-async then even name is incorrect `merge_branches_async_status`